lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -4023,6 +4024,90 @@ SPIRVValue *LLVMToSPIRVBase::transIntrinsicInst(IntrinsicInst *II,
                              transValue(II->getArgOperand(0), BB),
                              transValue(II->getArgOperand(1), BB), BB);
   }
+  case Intrinsic::vector_reduce_add:
+  case Intrinsic::vector_reduce_mul:
+  case Intrinsic::vector_reduce_and:
+  case Intrinsic::vector_reduce_or:
+  case Intrinsic::vector_reduce_xor: {
+    Op Op;
+    if (IID == Intrinsic::vector_reduce_add) {
```

Comment:
nit: switch statement may be better?

Reply:
If we aren't using fallthrough (which will be just jumps) in switch clause than it's just an equivalent of if clauses so the resulting code will be (should be) the same - branches or cmovs.